testnet-contracts/test/test_oracle.js

```diff
@@ -0,0 +1,386 @@
+        this.cosmosBridgeNonce,
+        this.hash,
+        [userOne, userThree],
+        sigV,
+        sigR,
+        sigS
+      ).should.be.fulfilled;
+    });
+
+    it("should emit an event upon successful prophecy processing", async function() {
+      const sigUserOne = parseSignature(this.userOneSignature);
+      const sigUserTwo = parseSignature(this.userTwoSignature);
+      const sigUserThree = parseSignature(this.userThreeSignature);
+
+      const sigV = [sigUserOne.v, sigUserTwo.v, sigUserThree.v];
+      const sigR = [sigUserOne.r, sigUserTwo.r, sigUserThree.r];
+      const sigS = [sigUserOne.s, sigUserTwo.s, sigUserThree.s];
+
+      const { logs } = await this.oracle.callProcessProphecyClaim(
+        this.cosmosBridgeNonce,
+        this.hash,
+        [userOne, userTwo, userThree],
+        sigV,
+        sigR,
+        sigS
+      );
+
+      const event = logs.find(e => e.event === "LogProphecyProcessed");
+
+      Number(event.args._cosmosBridgeClaimId).should.be.bignumber.equal(
+        this.cosmosBridgeNonce
+      );
+      Number(event.args._signedPower).should.be.bignumber.equal(
+        this.initialPowers[0] + this.initialPowers[1] + this.initialPowers[2]
+      );
+      Number(event.args._totalPower).should.be.bignumber.equal(this.totalPower);
+      event.args._submitter.should.be.equal(provider);
+    });
+
+    it("should not include the signatures of non-active validators", async function() {
+      // Generate signature from userFour (non-validator)
+      const userFourSignature = await web3.eth.sign(this.hash, accounts[4]);
+
+      const sigUserOne = parseSignature(this.userOneSignature);
+      const sigUserTwo = parseSignature(this.userTwoSignature);
+      const sigUserFour = parseSignature(userFourSignature);
+
+      const sigV = [sigUserOne.v, sigUserTwo.v, sigUserFour.v];
+      const sigR = [sigUserOne.r, sigUserTwo.r, sigUserFour.r];
+      const sigS = [sigUserOne.s, sigUserTwo.s, sigUserFour.s];
+
+      await this.oracle
+        .callProcessProphecyClaim(
+          this.cosmosBridgeNonce,
+          this.hash,
+          [userOne, userTwo, sigUserFour],
+          sigV,
+          sigR,
+          sigS
+        )
+        .should.be.rejectedWith("invalid address");
+    });
+
+    it("should not allow validator signatures to be applied twice", async function() {
+      const sigUserOne = parseSignature(this.userOneSignature);
+      const sigUserTwo = parseSignature(this.userTwoSignature);
+
+      const sigV = [sigUserOne.v, sigUserTwo.v, sigUserTwo.v];
+      const sigR = [sigUserOne.r, sigUserTwo.r, sigUserTwo.r];
+      const sigS = [sigUserOne.s, sigUserTwo.s, sigUserTwo.s];
+
+      await this.oracle
+        .callProcessProphecyClaim(
+          this.cosmosBridgeNonce,
+          this.hash,
+          [userOne, userTwo, userTwo],
+          sigV,
+          sigR,
+          sigS
+        )
+        .should.be.rejectedWith("does not meet the threshold");
+    });
+  });
+});
+
```

Comment:
need test+functionality for failed prophecies where consensus becomes impossible

Reply:
These tests will be included in the upcoming dynamic Valset PR, as consensus becoming impossible is linked to changes in the validator set.